extensions/google-javalite/src/main/kotlin/com/connectrpc/extensions/GoogleLiteProtoAdapter.kt

```diff
@@ -17,51 +17,52 @@ package com.connectrpc.extensions
 import com.connectrpc.CODEC_NAME_PROTO
 import com.connectrpc.Codec
 import com.google.protobuf.CodedOutputStream
-import com.google.protobuf.GeneratedMessageLite
+import com.google.protobuf.ExtensionRegistryLite
 import com.google.protobuf.Internal
 import com.google.protobuf.MessageLite
 import okio.Buffer
 import okio.BufferedSource
-import java.io.IOException
 import kotlin.reflect.KClass
+import kotlin.reflect.cast
 
 /**
  * Adapter to use Google's protobuf-javalite runtime for
  * deserializing and serializing data types.
  */
-internal class GoogleLiteProtoAdapter<E : GeneratedMessageLite<out E, *>>(
-    clazz: KClass<E>,
+internal class GoogleLiteProtoAdapter<E : MessageLite>(
+    private val clazz: KClass<E>,
+    private val registry: ExtensionRegistryLite,
 ) : Codec<E> {
     /**
      * Casting assumes the user is using Google's MessageLite type.
      */
-    @Suppress("UNCHECKED_CAST")
-    private val instance by lazy {
-        Internal.getDefaultInstance(clazz.java as Class<MessageLite>) as E
-    }
+    private val instance: E
+        get() = Internal.getDefaultInstance(clazz.java)
 
     override fun encodingName(): String {
         return CODEC_NAME_PROTO
     }
 
     override fun deserialize(source: BufferedSource): E {
-        return instance.parserForType.parseFrom(source.inputStream())
+        return clazz.cast(instance.parserForType.parseFrom(source.inputStream(), registry))
     }
 
     override fun serialize(message: E): Buffer {
-        return Buffer().write(message.toByteArray())
+        return serialize(message, false)
     }
 
     override fun deterministicSerialize(message: E): Buffer {
-        return try {
-            val result = Buffer()
-            val output = CodedOutputStream.newInstance(result.outputStream())
+        return serialize(message, true)
+    }
+
+    private fun serialize(message: E, deterministic: Boolean): Buffer {
```

Comment:
Same remark as non-lite java codec -- serious bugs in the deterministic flow, so I consolidated them so the deterministic special-case is more trivially correct.